src/server/database/database_provider.py

```diff
@@ -62,56 +51,57 @@ def handler(cls) -> Generator[DatabaseHandler, None, None]:
             DatabaseHandler: Handler consisting of cursor and query execution status (Message).
 
         """
-        with cls._try_get_cursor() as cursor:
-            if cursor is not None:
-                handler: DatabaseHandler = DatabaseHandler(cursor, Message.OK)
-                try:
-                    yield handler
-                except sqlite3.IntegrityError:
-                    handler.message = Message.INVALID_SQL
-                except sqlite3.ProgrammingError:
-                    handler.message = Message.INVALID_BINDINGS
-                except sqlite3.Error:
-                    handler.message = Message.UNKNOWN_ERROR
-                else:
-                    handler.message = Message.OK
-
-    @classmethod
-    @contextmanager
-    def _try_get_cursor(cls) -> Generator[Optional[sqlite3.Cursor], None, None]:
-        cursor: Optional[sqlite3.Cursor] = None
+        handler: DatabaseHandler = DatabaseHandler(None, Message.OK)  # type: ignore
         try:
-            cursor = cls.connection.cursor()
-            yield cursor
-        except sqlite3.Error as err:
-            print(f"ERROR: sqlite3.Cursor creation -- {err}")
-            yield None
-        finally:
-            if cursor is not None:
-                print("DEBUG: closing sqlite3.Cursor")
-                cursor.close()
-                print("DEBUG: committing sqlite3 changes")
-                cls.connection.commit()
+            cursor: psycopg.Cursor = cls.connection.cursor()
+            handler._cursor = cursor  # pylint: disable=W0212
+            yield handler
+        except psycopg.IntegrityError as err:
+            print(f"ERROR: {err}")
+            cls.connection.rollback()
+            handler.message = Message.DATABASE_INTEGRITY_ERROR
+        except psycopg.ProgrammingError as err:
+            print(f"ERROR: {err}")
+            cls.connection.rollback()
+            handler.message = Message.INVALID_SQL_QUERY
+        except psycopg.DataError as err:
+            print(f"ERROR: {err}")
+            cls.connection.rollback()
+            handler.message = Message.INVALID_SQL_QUERY
+        except psycopg.InternalError as err:
+            print(f"ERROR: {err}")
+            cls.connection.rollback()
+            handler.message = Message.INTERNAL_DATABASE_ERROR
+        except psycopg.NotSupportedError as err:
+            print(f"ERROR: {err}")
+            cls.connection.rollback()
+            handler.message = Message.UNSUPPORTED_OPERATION
+        except psycopg.errors.PipelineAborted as err:
+            print(f"ERROR: {err}")
+            handler.message = Message.NO_CONNECTION
+        except psycopg.Error as err:
+            print(f"ERROR: {err}")
+            cls.connection.rollback()
+            handler.message = Message.UNKNOWN_ERROR
```

Comment:
Every except block does the same, only message changes. Please get rid of duplicated code.

Reply:
Yes